pkg/cli/startbuild/startbuild_test.go

```diff
@@ -390,3 +395,156 @@ func TestStreamBuildLogs(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildConfigWithMissingResources(t *testing.T) {
```

Comment:
I would like to see more test cases here - there are a lot of permutations we should check:

1. BuildConfig with no Secrets or ConfigMaps
2. BuildConfig with build input Secrets only (available and missing)
3. BuildConfig with build input ConfigMaps only (available and missing)
4. BuildConfig with build input Secrets and ConfigMaps (available and missing)
5. BuildConfig with build volume Secrets in source strategy (available and missing)
6. BuildConfig with build volume ConfigMaps in source strategy (available and missing)
7. BuildConfig with build volume secrets in Docker strategy (available and missing)
8. BuildConfig with build volume ConfigMaps in Docker strategy (available and missing)
9. BuildConfig with build volume Secrets and ConfigMaps in Docker strategy (available and missing)
10. BuildConfig with build input and volumes - Secrets and ConfigMaps (basically everything)

I would be OK with combining some of these, as you would hit 19 test cases here doing each of these individually.